### ISSUE TITLE
Updating from 2.9.6 to 2.9.11 results in OPC Publisher unable to connect to kepserver opc server

### ISSUE THREAD

**Issue by MattCosturos** [Azure/Industrial-IoT#2345]
We have been using 2.9.6 of OPC Publisher.
Trying to update to use 2.9.11

The 2.9.11 version is unable to connect to our server

Update deployment to use 2.9.11 instead of 2.9.6

**Expected behavior**
Based on the minor version changes, I expect my existing configuration to work.  I expect all the existing config options and direct methods to work the same.

Logs here.  The logs seem to indicate the opc publisher module knows to use security (NotNone) but the Sender Cert is none. Is this the problem?

```

 ██████╗ ██████╗  ██████╗    ██████╗ ██╗   ██╗██████╗ ██╗     ██╗███████╗██╗  ██╗███████╗██████╗
██╔═══██╗██╔══██╗██╔════╝    ██╔══██╗██║   ██║██╔══██╗██║     ██║██╔════╝██║  ██║██╔════╝██╔══██╗
██║   ██║██████╔╝██║         ██████╔╝██║   ██║██████╔╝██║     ██║███████╗███████║█████╗  ██████╔╝
██║   ██║██╔═══╝ ██║         ██╔═══╝ ██║   ██║██╔══██╗██║     ██║╚════██║██╔══██║██╔══╝  ██╔══██╗
╚██████╔╝██║     ╚██████╗    ██║     ╚██████╔╝██████╔╝███████╗██║███████║██║  ██║███████╗██║  ██║
 ╚═════╝ ╚═╝      ╚═════╝    ╚═╝      ╚═════╝ ╚═════╝ ╚══════╝╚═╝╚══════╝╚═╝  ╚═╝╚══════╝╚═╝  ╚═╝
                                  2.9.11.2+63bedd6b75 (.NET 8.0.8/linux-x64/OPC Stack 1.5.374.78)

[24-09-18 20:37:28.0516] dbug: Furly.Azure.IoT.Edge.Services.IoTEdgeHubSdkClient[0]
      Connecting all clients to VM_HOST_NAME using AmqpOverTcp.
[24-09-18 20:37:28.0950] info: Furly.Azure.IoT.Edge.Services.IoTEdgeHubSdkClient[0]
      Running in iotedge context.
[24-09-18 20:37:28.6833] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[512]
      Imported the PFX private key for [032FA5E1751BCB839601370A8EB925744FFE2F72].
[24-09-18 20:37:28.7028] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaApplication[0]
      Own certificate Subject 'CN=Microsoft.Azure.IIoT, C=DE, S=Bav, O=Microsoft, DC=VM_HOST_NAME' (Thumbprint: 032FA5E1751BCB839601370A8EB925744FFE2F72) loaded.
[24-09-18 20:37:28.7048] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      Checking application instance certificate.
[24-09-18 20:37:28.7393] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[512]
      Imported the PFX private key for [032FA5E1751BCB839601370A8EB925744FFE2F72].
[24-09-18 20:37:28.7406] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      Check certificate: [CN=Microsoft.Azure.IIoT, C=DE, S=Bav, O=Microsoft, DC=VM_HOST_NAME] [032FA5E1751BCB839601370A8EB925744FFE2F72]
[24-09-18 20:37:28.7424] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      Check application instance certificate. [CN=Microsoft.Azure.IIoT, C=DE, S=Bav, O=Microsoft, DC=VM_HOST_NAME] [032FA5E1751BCB839601370A8EB925744FFE2F72]
[24-09-18 20:37:28.9586] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      Using the ApplicationUri: urn:VM_HOST_NAME:Microsoft.Azure.IIoT:microsoft:
[24-09-18 20:37:28.9647] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaApplication[0]
      Application own certificate store contains 2 certs.
[24-09-18 20:37:28.9673] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaApplication[0]
      01: Subject 'CN=Microsoft.Azure.IIoT, C=DE, S=Bav, O=Microsoft, DC=9150:da7c:87d7:e037:df54:4e95:b6b3:c1d4' (Thumbprint: 903C1D0C90EB1D762AF91CC45D5B7DAEEEB07695)
[24-09-18 20:37:28.9674] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaApplication[0]
      02: Subject 'CN=Microsoft.Azure.IIoT, C=DE, S=Bav, O=Microsoft, DC=VM_HOST_NAME' (Thumbprint: 032FA5E1751BCB839601370A8EB925744FFE2F72)
[24-09-18 20:37:28.9677] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaApplication[0]
      Trusted issuer store contains 0 certs.
[24-09-18 20:37:28.9682] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaApplication[0]
      Trusted issuer store has 0 CRLs.
[24-09-18 20:37:28.9690] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaApplication[0]
      Trusted peer store contains 3 certs.
[24-09-18 20:37:28.9692] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaApplication[0]
      01: Subject 'CN=Microsoft.Azure.IIoT, C=DE, S=Bav, O=Microsoft, DC=9150:da7c:87d7:e037:df54:4e95:b6b3:c1d4' (Thumbprint: 903C1D0C90EB1D762AF91CC45D5B7DAEEEB07695)
[24-09-18 20:37:28.9693] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaApplication[0]
      02: Subject 'CN=KEPServerEX/UA Server, O=Unknown, C=US, DC=USEA1SVKEP01.corp.mooginc.com' (Thumbprint: A8D8276CB251465002F2E30E2636BD942084BBA7)
[24-09-18 20:37:28.9694] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaApplication[0]
      03: Subject 'CN=Microsoft.Azure.IIoT, C=DE, S=Bav, O=Microsoft, DC=VM_HOST_NAME' (Thumbprint: 032FA5E1751BCB839601370A8EB925744FFE2F72)
[24-09-18 20:37:28.9694] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaApplication[0]
      Trusted peer store has 0 CRLs.
[24-09-18 20:37:28.9696] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaApplication[0]
      Rejected certificate store contains 0 certs.
[24-09-18 20:37:28.9955] dbug: Microsoft.Extensions.Hosting.Internal.Host[1]
      Hosting starting
[24-09-18 20:37:29.0280] warn: Microsoft.AspNetCore.DataProtection.Repositories.FileSystemXmlRepository[60]
      Storing keys in a directory '/root/.aspnet/DataProtection-Keys' that may not be persisted outside of the container. Protected data will be unavailable when container is destroyed. For more information go to https://aka.ms/aspnet/dataprotectionwarning
[24-09-18 20:37:29.0296] info: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[62]
      User profile is available. Using '/root/.aspnet/DataProtection-Keys' as key repository; keys will not be encrypted at rest.
[24-09-18 20:37:29.1822] dbug: Microsoft.AspNetCore.DataProtection.Repositories.FileSystemXmlRepository[37]
      Reading data from file '/root/.aspnet/DataProtection-Keys/key-95492018-6823-4a09-be2a-fca0c5923886.xml'.
[24-09-18 20:37:29.2228] dbug: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[18]
      Found key {95492018-6823-4a09-be2a-fca0c5923886}.
[24-09-18 20:37:29.2436] info: Azure.IIoT.OpcUa.Publisher.Services.PublisherModule[0]
      0: Module opc-publisher-1 reconnected due to Connection_Ok.
[24-09-18 20:37:29.2529] info: Azure.IIoT.OpcUa.Publisher.Services.PublisherModule[0]
      0: Module opc-publisher-1 opened.
[24-09-18 20:37:29.2542] dbug: Microsoft.AspNetCore.DataProtection.KeyManagement.DefaultKeyResolver[13]
      Considering key {95492018-6823-4a09-be2a-fca0c5923886} with expiration date 2024-12-17 20:29:00Z as default key.
[24-09-18 20:37:29.2610] dbug: Microsoft.AspNetCore.DataProtection.TypeForwardingActivator[0]
      Forwarded activator type request from Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ConfigurationModel.AuthenticatedEncryptorDescriptorDeserializer, Microsoft.AspNetCore.DataProtection, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60 to Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ConfigurationModel.AuthenticatedEncryptorDescriptorDeserializer, Microsoft.AspNetCore.DataProtection, Culture=neutral, PublicKeyToken=adb9793829ddae60
[24-09-18 20:37:29.2647] dbug: Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ManagedAuthenticatedEncryptorFactory[11]
      Using managed symmetric algorithm 'System.Security.Cryptography.Aes'.
[24-09-18 20:37:29.2650] dbug: Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ManagedAuthenticatedEncryptorFactory[10]
      Using managed keyed hash algorithm 'System.Security.Cryptography.HMACSHA256'.
[24-09-18 20:37:29.2718] dbug: Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingProvider[2]
      Using key {95492018-6823-4a09-be2a-fca0c5923886} as the default key.
[24-09-18 20:37:29.2734] dbug: Microsoft.AspNetCore.DataProtection.Internal.DataProtectionHostedService[65]
      Key ring with default key {95492018-6823-4a09-be2a-fca0c5923886} was loaded during application startup.
[24-09-18 20:37:29.3008] dbug: Furly.Azure.IoT.Edge.Services.IoTEdgeTwinClient[0]
      Initialize device twin ...
[24-09-18 20:37:29.3997] dbug: Microsoft.AspNetCore.Mvc.ModelBinding.ModelBinderFactory[12]
      Registered model binder providers, in the following order: Asp.Versioning.ApiVersionModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.BinderTypeModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.ServicesModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.BodyModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.HeaderModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.FloatingPointTypeModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.EnumTypeModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.DateTimeModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.SimpleTypeModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.TryParseModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.CancellationTokenModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.ByteArrayModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.FormFileModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.FormCollectionModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.KeyValuePairModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.DictionaryModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.ArrayModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.CollectionModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.ComplexObjectModelBinderProvider
[24-09-18 20:37:29.6598] dbug: Furly.Azure.IoT.Edge.Services.IoTEdgeTwinClient[0]
      Device twin initialized successfully.
[24-09-18 20:37:29.6619] dbug: Furly.Azure.IoT.Edge.Services.IoTEdgeTwinClient[0]
      Initialize device twin ...
[24-09-18 20:37:29.6947] warn: Microsoft.AspNetCore.Server.Kestrel[0]
      Overriding address(es) 'http://*:8080'. Binding to endpoints defined via IConfiguration and/or UseKestrel() instead.
[24-09-18 20:37:29.7118] dbug: Microsoft.AspNetCore.Server.Kestrel[0]
      Failed to bind to http://[::]:80 (IPv6Any). Attempting to bind to http://0.0.0.0:80 instead.
[24-09-18 20:37:29.7177] info: Microsoft.Hosting.Lifetime[14]
      Now listening on: http://0.0.0.0:80
[24-09-18 20:37:29.7178] info: Microsoft.Hosting.Lifetime[14]
      Now listening on: https://0.0.0.0:443
[24-09-18 20:37:29.7179] dbug: Microsoft.AspNetCore.Hosting.Diagnostics[13]
      Loaded hosting startup assembly Azure.IIoT.OpcUa.Publisher.Module
[24-09-18 20:37:29.7285] dbug: Furly.Azure.IoT.Edge.Services.IoTEdgeTwinClient[0]
      Device twin initialized successfully.
[24-09-18 20:37:29.7335] info: Azure.IIoT.OpcUa.Publisher.Services.PublisherConfigurationService[0]
      Found published Nodes File with hash 8EB4C9CEFF19C358C0C40D604CAE3161F115802D83389DEFC71F74685435262D, loading...
[24-09-18 20:37:29.7341] dbug: Azure.IIoT.OpcUa.Publisher.Storage.PublishedNodesConverter[0]
      Reading and validating published nodes file...
[24-09-18 20:37:29.7819] info: Azure.IIoT.OpcUa.Publisher.Storage.PublishedNodesConverter[0]
      Read 1 entry models from published nodes file in 00:00:00.0476767
[24-09-18 20:37:29.8368] info: Azure.IIoT.OpcUa.Publisher.Discovery.NetworkDiscovery[0]
      Starting discovery processor...
[24-09-18 20:37:29.8400] info: Azure.IIoT.OpcUa.Publisher.Storage.PublishedNodesConverter[0]
      Converted published nodes entry models to jobs in 00:00:00.0567134
[24-09-18 20:37:29.8401] info: Azure.IIoT.OpcUa.Publisher.Services.PublisherConfigurationService[0]
      Refreshing publisher configuration completed.
[24-09-18 20:37:29.8558] info: Azure.IIoT.OpcUa.Publisher.Services.PublisherDiagnosticCollector[0]
      Tracking diagnostics for fb43bc4a193fbd53c26cf4a64f529e3f89108c78 was (re-)started.
[24-09-18 20:37:29.8990] info: Azure.IIoT.OpcUa.Publisher.Services.NetworkMessageSink[0]
      Writer group myGroup set up to publish notifications as soon as they arrive and individually with at most 252 kb to IoTEdge with unknown layout and Json encoding (queuing at most 4096 subscription notifications)...
[24-09-18 20:37:29.9184] dbug: Azure.IIoT.OpcUa.Publisher.Services.WriterGroupDataSource[0]
      Open new writer (null) with subscription opc.tcp://myServer:myPort_457DFB59_myGroup:f31ab2178ae2416edab71d726d0ee7a42af0c8d4_0 in writer group fb43bc4a193fbd53c26cf4a64f529e3f89108c78...
[24-09-18 20:37:29.9288] info: Azure.IIoT.OpcUa.Publisher.Services.WriterGroupDataSource[0]
      Writer with subscription opc.tcp://myServer:myPort_457DFB59_myGroup:f31ab2178ae2416edab71d726d0ee7a42af0c8d4_0 in writer group fb43bc4a193fbd53c26cf4a64f529e3f89108c78 new subscription received.
[24-09-18 20:37:29.9505] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClientManager[0]
      New client opc.tcp://myServer:myPort_457DFB59_myGroup [state:Disconnected|refs:0] created.
[24-09-18 20:37:29.9528] dbug: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaSubscription[0]
      Trigger management of offline subscription f31ab2178ae2416edab71d726d0ee7a42af0c8d4_0:0...
[24-09-18 20:37:29.9539] dbug: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      opc.tcp://myServer:myPort_457DFB59_myGroup [state:Disconnected|refs:1]: Processing event Connect in State Disconnected...
[24-09-18 20:37:29.9543] info: Azure.IIoT.OpcUa.Publisher.Services.WriterGroupDataSource[0]
      New writer with subscription opc.tcp://myServer:myPort_457DFB59_myGroup:f31ab2178ae2416edab71d726d0ee7a42af0c8d4_0 in writer group fb43bc4a193fbd53c26cf4a64f529e3f89108c78 opened.
[24-09-18 20:37:29.9583] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      opc.tcp://myServer:myPort_457DFB59_myGroup [state:Connecting|refs:1]: Session opc.tcp://myServer:myPort_457DFB59_myGroup with opc.tcp://myServer:myPort changed from Disconnected to Connecting
[24-09-18 20:37:29.9584] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      opc.tcp://myServer:myPort_457DFB59_myGroup [state:Connecting|refs:1]: Connecting to opc.tcp://myServer:myPort...
[24-09-18 20:37:29.9604] info: Azure.IIoT.OpcUa.Publisher.Services.PublisherModule[0]
      Starting OpcPublisher module version 2.9.11.2...
[24-09-18 20:37:29.9710] info: Azure.IIoT.OpcUa.Publisher.Services.RuntimeStateReporter[0]
      Api Key exists in IoTEdge store...
[24-09-18 20:37:30.0052] info: Azure.IIoT.OpcUa.Publisher.Services.RuntimeStateReporter[0]
      Using valid Certificate found in IoTEdge store (renewal in 24.23:10:28.0076488)...
[24-09-18 20:37:30.0054] info: Azure.IIoT.OpcUa.Publisher.Services.PublisherModule[0]
      OpcPublisher module version 2.9.11.2 started.
[24-09-18 20:37:30.0084] info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
[24-09-18 20:37:30.0085] info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Production
[24-09-18 20:37:30.0085] info: Microsoft.Hosting.Lifetime[0]
      Content root path: /app
[24-09-18 20:37:30.0086] dbug: Microsoft.Extensions.Hosting.Internal.Host[2]
      Hosting started
[24-09-18 20:37:30.1177] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      ChannelId 0: Token #0 created. CreatedAt=20:37:30.116. Lifetime=3600000.
[24-09-18 20:37:30.1331] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      Security Policy: http://opcfoundation.org/UA/SecurityPolicy#None
[24-09-18 20:37:30.1332] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      Sender Certificate: (none)
[24-09-18 20:37:30.1419] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      ChannelId 3389307052: Token #1 activated. CreatedAt=20:37:30.116. Lifetime=3600000.
[24-09-18 20:37:30.1431] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      ChannelId 3389307052: Token Expiry 09/18/2024 21:37:30, renewal scheduled in 2699980 ms.
[24-09-18 20:37:30.1609] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      opc.tcp://myServer:myPort_457DFB59_myGroup [state:Connecting|refs:1]: Discovery endpoint opc.tcp://myServer:myPort/ returned endpoints. Selecting endpoint opc.tcp://myServer:myPort/ with SecurityMode NotNone and any SecurityPolicyUri from:
            #051: opc.tcp://myServer:myPort/|SignAndEncrypt [http://opcfoundation.org/UA/SecurityPolicy#Basic256Sha256]
[24-09-18 20:37:30.1625] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      opc.tcp://myServer:myPort_457DFB59_myGroup [state:Connecting|refs:1]: Endpoint #051: opc.tcp://myServer:myPort/|SignAndEncrypt [http://opcfoundation.org/UA/SecurityPolicy#Basic256Sha256] selected!
[24-09-18 20:37:30.6072] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      ChannelId 3389307052: CLIENTCHANNEL SOCKET CLOSED: 00E04189
[24-09-18 20:37:30.6135] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      #1 - opc.tcp://myServer:myPort_457DFB59_myGroup [state:Connecting|refs:1]: Creating session opc.tcp://myServer:myPort_457DFB59_myGroup with endpoint opc.tcp://myServer:myPort/...
[24-09-18 20:37:30.6510] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      ChannelId 0: Token #0 created. CreatedAt=20:37:30.650. Lifetime=3600000.
[24-09-18 20:37:30.6644] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      ChannelId 0: CLIENTCHANNEL SOCKET CLOSED: 02EC8B68
[24-09-18 20:37:30.6658] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaSession[0]
      : Session disposed.
[24-09-18 20:37:30.6682] dbug: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      opc.tcp://myServer:myPort_457DFB59_myGroup [state:Connecting|refs:1]: BadNoCommunication 'Error establishing a connection: Error received from remote host: An error occurred verifying security.' => Error
[24-09-18 20:37:30.6683] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      opc.tcp://myServer:myPort_457DFB59_myGroup [state:Error|refs:1]: Session opc.tcp://myServer:myPort_457DFB59_myGroup with opc.tcp://myServer:myPort changed from Connecting to Error
[24-09-18 20:37:30.6683] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      #2 - opc.tcp://myServer:myPort_457DFB59_myGroup [state:Error|refs:1]: Failed to connect to opc.tcp://myServer:myPort/: Error establishing a connection: Error received from remote host: An error occurred verifying security....
[24-09-18 20:37:30.6688] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      opc.tcp://myServer:myPort_457DFB59_myGroup [state:Error|refs:1]: Retrying connecting session in 00:00:01...
[24-09-18 20:37:30.6690] dbug: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      Connect: Event opc.tcp://myServer:myPort_457DFB59_myGroup [state:Error|refs:1] in State Connecting processed.
[24-09-18 20:37:30.6691] dbug: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      opc.tcp://myServer:myPort_457DFB59_myGroup [state:Error|refs:1]: Processing event SubscriptionManage in State Connecting...
[24-09-18 20:37:30.6695] dbug: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      SubscriptionManage: Event opc.tcp://myServer:myPort_457DFB59_myGroup [state:Error|refs:1] in State Connecting processed.
[24-09-18 20:37:31.6707] dbug: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      opc.tcp://myServer:myPort_457DFB59_myGroup [state:Error|refs:1]: Processing event ConnectRetry in State Connecting...
[24-09-18 20:37:31.6708] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      opc.tcp://myServer:myPort_457DFB59_myGroup [state:Connecting|refs:1]: Session opc.tcp://myServer:myPort_457DFB59_myGroup with opc.tcp://myServer:myPort changed from Error to Connecting
[24-09-18 20:37:31.6709] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      opc.tcp://myServer:myPort_457DFB59_myGroup [state:Connecting|refs:1]: Connecting to opc.tcp://myServer:myPort...
[24-09-18 20:37:31.6731] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      ChannelId 0: Token #0 created. CreatedAt=20:37:31.673. Lifetime=3600000.
[24-09-18 20:37:31.6736] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      Security Policy: http://opcfoundation.org/UA/SecurityPolicy#None
[24-09-18 20:37:31.6737] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      Sender Certificate: (none)
[24-09-18 20:37:31.6739] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      ChannelId 1096434167: Token #1 activated. CreatedAt=20:37:31.673. Lifetime=3600000.
[24-09-18 20:37:31.6740] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      ChannelId 1096434167: Token Expiry 09/18/2024 21:37:31, renewal scheduled in 2699999 ms.
[24-09-18 20:37:31.6767] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      opc.tcp://myServer:myPort_457DFB59_myGroup [state:Connecting|refs:1]: Discovery endpoint opc.tcp://myServer:myPort/ returned endpoints. Selecting endpoint opc.tcp://myServer:myPort/ with SecurityMode NotNone and any SecurityPolicyUri from:
            #051: opc.tcp://myServer:myPort/|SignAndEncrypt [http://opcfoundation.org/UA/SecurityPolicy#Basic256Sha256]
[24-09-18 20:37:31.6769] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      opc.tcp://myServer:myPort_457DFB59_myGroup [state:Connecting|refs:1]: Endpoint #051: opc.tcp://myServer:myPort/|SignAndEncrypt [http://opcfoundation.org/UA/SecurityPolicy#Basic256Sha256] selected!
[24-09-18 20:37:32.1737] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      ChannelId 1096434167: CLIENTCHANNEL SOCKET CLOSED: 0346AEFF
[24-09-18 20:37:32.1738] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      #1 - opc.tcp://myServer:myPort_457DFB59_myGroup [state:Connecting|refs:1]: Creating session opc.tcp://myServer:myPort_457DFB59_myGroup with endpoint opc.tcp://myServer:myPort/...
[24-09-18 20:37:32.1820] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      ChannelId 0: Token #0 created. CreatedAt=20:37:32.181. Lifetime=3600000.
[24-09-18 20:37:32.1862] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaStack[0]
      ChannelId 0: CLIENTCHANNEL SOCKET CLOSED: 030E9943
[24-09-18 20:37:32.1866] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaSession[0]
      : Session disposed.
[24-09-18 20:37:32.1870] dbug: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      opc.tcp://myServer:myPort_457DFB59_myGroup [state:Connecting|refs:1]: BadNoCommunication 'Error establishing a connection: Error received from remote host: An error occurred verifying security.' => Error
[24-09-18 20:37:32.1871] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      opc.tcp://myServer:myPort_457DFB59_myGroup [state:Error|refs:1]: Session opc.tcp://myServer:myPort_457DFB59_myGroup with opc.tcp://myServer:myPort changed from Connecting to Error
[24-09-18 20:37:32.1872] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      #2 - opc.tcp://myServer:myPort_457DFB59_myGroup [state:Error|refs:1]: Failed to connect to opc.tcp://myServer:myPort/: Error establishing a connection: Error received from remote host: An error occurred verifying security....
[24-09-18 20:37:32.1873] info: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      opc.tcp://myServer:myPort_457DFB59_myGroup [state:Error|refs:1]: Retrying connecting session in 00:00:01.0320000...
[24-09-18 20:37:32.1874] dbug: Azure.IIoT.OpcUa.Publisher.Stack.Services.OpcUaClient[0]
      ConnectRetry: Event opc.tcp://myServer:myPort_457DFB59_myGroup [state:Error|refs:1] in State Connecting processed.

```

**Comment by marcschier** [comment]
@MattCosturos, it looks like the server did not trust the OPC publisher certificate, is it in it's trust list after reinit?  The logs of kepserver should provide indication why it rejected the certificate, can you investigate to get the root cause from the kepserver logs?

**Comment by MattCosturos** [comment]
@marcschier  - Yup, that was it.
Our deployment is setup to mount a volume to our root os, and the PKI certs are saved there.
However it looks like doing a version update from 2.9.6 to 2.9.11 generated new certs. Is that expected behavior?

If I stop iotedge, delete the container, and restart iotedge, the cert from the mounted volume is correctly loaded.

**Comment by marcschier** [comment]
We had some problems with the format of the cert not being backwards compatible to 2.8 in 2.9 and we corrected that after 2.9.7(?) but unfortunately that shoots down upgrades from previous versions.  To provide a. Workaround we introduced a switch to read the first valid cert in the own folder and use its info to configure the publisher (--cfa).  More info can be found in the readme.md pki section.
